packages/medusa/src/api/routes/store/product-tags/list-product-tags.ts

```diff
@@ -0,0 +1,181 @@
+ *     schema:
+ *       type: array
+ *       items:
+ *         type: string
+ *   - in: query
+ *     name: id
+ *     style: form
+ *     explode: false
+ *     description: The tag IDs to search for
+ *     schema:
+ *       type: array
+ *       items:
+ *         type: string
+ *   - (query) q {string} A query string to search values for
+ *   - in: query
+ *     name: created_at
+ *     description: Date comparison for when resulting product tags were created.
+ *     schema:
+ *       type: object
+ *       properties:
+ *         lt:
+ *            type: string
+ *            description: filter by dates less than this date
+ *            format: date
+ *         gt:
+ *            type: string
+ *            description: filter by dates greater than this date
+ *            format: date
+ *         lte:
+ *            type: string
+ *            description: filter by dates less than or equal to this date
+ *            format: date
+ *         gte:
+ *            type: string
+ *            description: filter by dates greater than or equal to this date
+ *            format: date
+ *   - in: query
+ *     name: updated_at
+ *     description: Date comparison for when resulting product tags were updated.
+ *     schema:
+ *       type: object
+ *       properties:
+ *         lt:
+ *            type: string
+ *            description: filter by dates less than this date
+ *            format: date
+ *         gt:
+ *            type: string
+ *            description: filter by dates greater than this date
+ *            format: date
+ *         lte:
+ *            type: string
+ *            description: filter by dates less than or equal to this date
+ *            format: date
+ *         gte:
+ *            type: string
+ *            description: filter by dates greater than or equal to this date
+ *            format: date
+ * x-codeSamples:
+ *   - lang: JavaScript
+ *     label: JS Client
+ *     source: |
+ *       import Medusa from "@medusajs/medusa-js"
+ *       const medusa = new Medusa({ baseUrl: MEDUSA_BACKEND_URL, maxRetries: 3 })
+ *       medusa.store.productTags.list()
+ *       .then(({ product_tags }) => {
+ *         console.log(product_tags.length);
+ *       });
+ *   - lang: Shell
+ *     label: cURL
+ *     source: |
+ *       curl --location --request GET 'https://medusa-url.com/store/product-tags'
+ * tags:
+ *   - Product Tag
+ * responses:
+ *  "200":
+ *    description: OK
+ *    content:
+ *      application/json:
+ *        schema:
+ *          type: object
+ *          properties:
+ *            product_tags:
+ *              $ref: "#/components/schemas/ProductTag"
+ *            count:
+ *              type: integer
+ *              description: The total number of items available
+ *            offset:
+ *              type: integer
+ *              description: The number of items skipped before these items
+ *            limit:
+ *              type: integer
```

Comment:
suggestion(non-blocking): create a schema for this return type

Reply:
Maybe I am missing something but there already is a ProductTag.yaml file in `docs/api/store/compoents`, do I need to add something else?

Reply:
definitely non-blocking, what I had in mind was a schema along the lines of what Pat has been defining since if we have multiple of these returning the same thing then we generate multiple types (hence the type hell we've looked at previously. I meant to suggest that for consistency we could define a schema like this where we've defined the list return type `StoreProductTagsListRes`: 
```
/**
 * @schema StoreProductTagsListRes
 * type: object
 * properties:
 *   product_tags:
 *    type: array
 *    items:
 *      $ref: "#/components/schemas/ProductTag"
 *   count:
 *     type: integer
 *     description: The total number of items available
 *   offset:
 *     type: integer
 *     description: The number of items skipped before these items
 *   limit:
 *     type: integer
 *     description: The number of items per page
 */
```

And referencing that from the oas definition instead of defining the return type there: 
```
 * responses:
 *   200:
 *     description: OK
 *     content:
 *       application/json:
 *         schema:
 *           $ref: "#/components/schemas/StoreProductTagsListRes"
``` 

Since we only use this here I've marked it as non-blocking 😄